packages/flutter_tools/lib/src/commands/update_packages.dart

```diff
@@ -103,168 +152,296 @@ class UpdatePackagesCommand extends FlutterCommand {
   }
 }
 
+// The various sections of a pubspec.yaml file. We care about the
+// "dependencies", "dev_dependencies", and "dependency_overrides" sections, the
+// others are all bucketed into "other".
 enum Section { dependencies, devDependencies, dependencyOverrides, other }
 
+// The various kinds of dependencies we know and care about.
+enum DependencyKind {
+  // Dependencies that will be path or sdk dependencies but
+  // for which we haven't yet parsed the data.
+  unknown,
+
+  // Regular dependencies with a specified version range.
+  normal,
+
+  // Dependency that uses an explicit path, e.g. into the Dart SDK.
+  path,
+
+  // Dependency defined as coming from an SDK (typically "sdk: flutter").
+  sdk,
+
+  // A dependency that was "normal", but for which we later found a "path" or
+  // "sdk" dependency in the dependency_overrides section.
+  overridden,
+}
+
+// This is the string we output next to each of our autogenerated transitive
+// dependencies so that we can ignore them the next time we parse the
+// pubspec.yaml file.
 const String kTransitiveMagicString = '# TRANSITIVE DEPENDENCY';
 
+// This class represents a pubspec.yaml file for the purposes of upgrading the
+// dependencies as done by this file.
 class PubspecYaml {
+  // You create one of these by providing a directory, from which we obtain the
+  // pubspec.yaml and parse it into a line-by-line form.
   factory PubspecYaml(Directory directory) {
     final File file = _pubspecFor(directory);
     return new PubspecYaml._(file, _parse(file.path, file.readAsLinesSync()));
   }
 
   PubspecYaml._(this.file, this.inputData);
 
-  final File file;
+  final File file; // The actual pubspec.yaml file.
 
-  final List<PubspecLine> inputData;
+  final List<PubspecLine> inputData; // Each line of the pubspec.yaml file, parsed(ish).
```

Comment:
Maybe also say that it does not contain unversioned dependencies (git, sdk, path)?

Reply:
It contains every line of the pubspec.yaml (some pairs of lines end up in a single object in this list, but that seems a bit detailed to list here).